### PR TITLE
Add jsdoc/godoc for Hash and Value

### PR DESCRIPTION
**Pull request by arv**
Towards #2297

### CODE REVIEW

File: go/hash/hash.go [hunk 1]
```diff
@@ -107,25 +121,32 @@ func Parse(s string) Hash {
...
 }
 
+// Less compares two hashes returning whether this Hash is less than other.
 func (r Hash) Less(other Hash) bool {
 	return bytes.Compare(r.digest[:], other.digest[:]) < 0
 }
 
+// Greater compares two hashes returning whether this Hash is greater than other.
 func (r Hash) Greater(other Hash) bool {
+	// TODO: Remove this
```
Comment (mikegray) [inline go/hash/hash.go]: Hmm, did you mean to add this comment here?

File: go/types/value.go [hunk 2]
```diff
@@ -8,15 +8,31 @@ import (
...
 )
 
-// Value is implemented by every noms value
+// Value is the interface all Noms values implement.
 type Value interface {
+	// Equals determines if two different Noms values represents the same underlying value.
 	Equals(other Value) bool
+
+	// Less determines if this Noms value is less than another Noms value.
+	// When comparing two Noms values and both are comparable and the same type (Bool, Number or
```
Comment (mikegray) [inline go/types/value.go]: Should we document the details of the implementation here or maybe just leave that to the documentation of the implementations of Less?

**Comment by mikegray** [comment]
lgtm with few open questions.

File: go/hash/hash.go (hunk 1 above)
Comment (arv) [inline go/hash/hash.go]: Yes. I didn't want to change the API in this PR.

Having Less and Greater is pretty silly. I'll open an issue.

File: go/types/value.go (hunk 2 above)
Comment (arv) [inline go/types/value.go]: I wanted this somewhere. It is not clear where it belongs though.

**Pull request merged**